vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py

```diff
@@ -0,0 +1,264 @@
+    url = url if url.startswith("http") else "https://" + url
+    if len(url.split(" ")) == 1:
+        return url
+    return None
+
+
+def _get_notes(site: dict) -> Optional[List[str]]:
+    if site["attributes"]["Instructions"]:
+        return [site["attributes"]["Instructions"]]
+
+    return None
+
+
+def _get_active(site: dict) -> Optional[bool]:
+    # end date may be important to check to determine if the site is historicle or current but i dont really feel like digging through the docs rn. see https://github.com/CAVaccineInventory/vaccine-feed-ingest/pull/119 for links that eventually lead to specs on the
+    # end_date = site["attributes"].get("end_date")
+
+    status = site["attributes"].get("status")
+
+    status_options = {
+        "Open": True,
+        "Closed": False,
+        "Testing Restricted": True,
+        "Scheduled to Open": False,
+        "Temporarily Closed": False,
+    }
+
+    return try_lookup(status_options, status, None, name="active status lookup")
+
+
+def _get_access(site: dict) -> Optional[List[str]]:
+    drive = site["attributes"].get("drive_through")
+    drive_bool = drive is not None
+
+    # walk = site["attributes"].get("drive_through")
+    # walk_bool = drive is not None
+
+    wheelchair = site["attributes"].get("Wheelchair_Accessible")
+
+    wheelchair_options = {
+        "Yes": "yes",
+        "Partially": "partial",
+        "Unknown": "no",
+        "Not Applicable": "no",
+        "NA": "no",
+    }
+    wheelchair_bool = try_lookup(
+        wheelchair_options, wheelchair, "no", name="wheelchair access"
+    )
+
+    return schema.Access(drive=drive_bool, wheelchair=wheelchair_bool)
+
+
+def try_lookup(mapping, value, default, name=None):
+    if value is None:
+        return default
+
+    try:
+        return mapping[value]
+    except KeyError as e:
+        name = " for " + name or ""
+        logger.error("value not present in lookup table%s: %s", name, e)
+
+        return default
+
+
+def _get_published_at(site: dict) -> Optional[str]:
+    date_with_millis = site["attributes"]["CreationDate"]
+    if date_with_millis:
+        date = datetime.datetime.fromtimestamp(date_with_millis / 1000)  # Drop millis
+        return date.isoformat()
+
+    return None
+
+
+def try_get_list(lis, index, default=None):
+    if lis is None:
+        return default
+
+    try:
+        value = lis[index]
+        if value == "none":
+            logger.warn("saw none value")
+        return value
+    except IndexError:
+        return default
+
+
+def _get_normalized_location(site: dict, timestamp: str) -> schema.NormalizedLocation:
+
+    addrsplit = site["attributes"]["fulladdr"].split(", ")
+
+    zip = site["attributes"]["fulladdr"][-5:]
+    zip = zip if zip.isnumeric() else None
+
+    city_state_zip = addrsplit[1].split(" ") if try_get_list(addrsplit, 1) else None
+
+    state = site["attributes"]["State"] or None
+    state = state.strip() if state is not None else None
```

Comment:
This looks good for potential follow-up!

Reply:
yep, this is more hacky than anything else. might be ripe to find an address-parsing library similar to my library for opening hours and use that instead

Reply:
https://github.com/datamade/usaddress looks good for this. should this wait for another PR to make this review easier?

Reply:
I'm going to merge this - let's add address improvements in a followup!